src/main/java/frc/robot/systems/ElevatorArmFSM.java

```diff
@@ -0,0 +1,298 @@
+package frc.robot.systems;
+
+// WPILib Imports
+
+// Third party Hardware Imports
+import com.revrobotics.CANSparkMax;
+
+// Robot Imports
+import frc.robot.TeleopInput;
+import frc.robot.HardwareMap;
+import com.revrobotics.SparkMaxPIDController;
+import com.revrobotics.SparkMaxLimitSwitch;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+public class ElevatorArmFSM {
+	/* ======================== Constants ======================== */
+	// FSM state definitions
+	public enum FSMState {
+		IDLE,
+		HIGH,
+		MIDDLE,
+		LOW,
+		MOVING,
+		ZEROING
+	}
+
+	private static final float UP_POWER = 0.1f;
+	private static final float DOWN_POWER = -0.1f;
+	private static final float ZEROING_POWER = -0.1f;
+	private static final double PID_CONSTANT_ARM_P = 0.005;
+	private static final double PID_CONSTANT_ARM_I = 0.00000001;
+	private static final double PID_CONSTANT_ARM_D = 0.00000001;
```

Comment:
Are the I and D actually doing anything?

Reply:
In this case, I and D have little effect because we're currently trusting the motors, which mainly involves the P constant. However, once we add weight, we anticipiate I and D to have a greater effect.